hoprd/rest-api/src/session.rs

```diff
@@ -0,0 +1,131 @@
+use std::sync::Arc;
+
+use axum::{
+    extract::{Json, State},
+    http::status::StatusCode,
+    response::IntoResponse,
+};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+
+use hopr_lib::{PathOptions, PeerId, SessionCapability, SessionClientConfig};
+use tokio::{io::copy_bidirectional_with_sizes, net::TcpListener};
+use tokio_util::compat::FuturesAsyncReadCompatExt;
+use tracing::{error, info};
+
+use crate::{ApiErrorStatus, InternalState, BASE_PATH};
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[schema(example = json!({
+        "destination": "12D3KooWR4uwjKCDCAY1xsEFB4esuWLF9Q5ijYvCjz5PNkTbnu33",
+        "path": {
+            "Hops": 1
+        },
+        "capabilities": ["Segmentation", "Retransmission"]
+    }))]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionClientRequest {
+    #[serde_as(as = "DisplayFromStr")]
+    pub destination: PeerId,
+    pub path: PathOptions,
+    pub capabilities: Vec<SessionCapability>,
+    pub port: Option<u16>,
+}
+
+impl From<SessionClientRequest> for SessionClientConfig {
+    fn from(value: SessionClientRequest) -> Self {
+        Self {
+            peer: value.destination,
+            path_options: value.path,
+            capabilities: value.capabilities,
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[schema(example = json!({
+        "port": 5542,
+    }))]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionClientResponse {
+    pub port: u16,
+}
+
+/// Creates a new client session returing a dedicated session listening port.
```

Comment:
I'm unclear on how this endpoint is supposed to be used, could you elaborate in the docs?